crypto/fipsmodule/kdf/sskdf.c

```diff
@@ -0,0 +1,314 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR ISC
+
+#include <assert.h>
+#include <openssl/base.h>
+#include <openssl/digest.h>
+#include <openssl/hmac.h>
+#include <openssl/kdf.h>
+#include "../delocate.h"
+#include "internal.h"
+
+static int SSKDF_VARIANT_DIGEST_CTX_init(SSKDF_VARIANT_CTX *ctx,
+                                         const EVP_MD *digest) {
+  SSKDF_VARIANT_DIGEST_CTX *variant_ctx = NULL;
+  EVP_MD_CTX *md_ctx = NULL;
+
+  int ret = 0;
+
+  if (!ctx || !digest) {
+    goto err;
+  }
+
+  variant_ctx = OPENSSL_malloc(sizeof(SSKDF_VARIANT_DIGEST_CTX));
+  if (!variant_ctx) {
+    goto err;
+  }
+
+  md_ctx = EVP_MD_CTX_new();
+  if (!md_ctx) {
+    goto err;
+  }
+
+  ret = 1;
+  variant_ctx->digest = digest;
+  variant_ctx->ctx = md_ctx;
+  ctx->data = variant_ctx;
+
+  return ret;
```

Comment:
nit: you can delete this return.

Reply:
That would then cause the structures to be unallcoated below, which is not the desired behavior.